### PR TITLE
SUBMARINE-698. Enable notebook to associate with git-based repo

### PR DESCRIPTION
**Pull request by aeioulisa**
### What is this PR for?
Switch to JupyterLab from jupyter-notebook

### What type of PR is it?
[Improvement]

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/SUBMARINE-698

### How should this be tested?
https://travis-ci.org/github/aeioulisa/submarine/builds/756075486

### Screenshots (if appropriate)
image

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: dev-support/docker-images/jupyter/Dockerfile [hunk 1]
```diff
@@ -92,12 +92,14 @@ RUN wget --quiet https://repo.continuum.io/miniconda/Miniconda3-py37_${MINICONDA
...
-    cp -r submarine/submarine-sdk/pysubmarine/example/{data,deepfm_example.ipynb,deepfm.json} $HOME && \
-    rm submarine -rf
+RUN git clone https://github.com/apache/submarine && \
+pip install submarine/submarine-sdk/pysubmarine && \
+conda install nodejs && \
+conda install -c conda-forge jupyterlab jupyterlab-git && \
+jupyter lab build && \
+cp submarine/submarine-sdk/pysubmarine/example/submarine_experiment_sdk.ipynb $HOME && \
+cp -r submarine/submarine-sdk/pysubmarine/example/{data,deepfm_example.ipynb,deepfm.json} $HOME && \
+rm submarine -rf
```
Comment (pingsutw) [inline dev-support/docker-images/jupyter/Dockerfile:102]:
```suggestion
    pip install submarine/submarine-sdk/pysubmarine && \
    conda install nodejs && \
    conda install -c conda-forge jupyterlab jupyterlab-git && \
    jupyter lab build && \
    cp submarine/submarine-sdk/pysubmarine/example/submarine_experiment_sdk.ipynb $HOME && \
    cp -r submarine/submarine-sdk/pysubmarine/example/{data,deepfm_example.ipynb,deepfm.json} $HOME && \
    rm submarine -rfpip install submarine/submarine-sdk/pysubmarine && \
    conda install nodejs && \
    conda install -c conda-forge jupyterlab jupyterlab-git && \
    jupyter lab build && \
    cp submarine/submarine-sdk/pysubmarine/example/submarine_experiment_sdk.ipynb $HOME && \
    cp -r submarine/submarine-sdk/pysubmarine/example/{data,deepfm_example.ipynb,deepfm.json} $HOME && \
    rm submarine -rf
```

Comment (pingsutw) [reply dev-support/docker-images/jupyter/Dockerfile:102]: Btw, Why do we need to install `nodejs`? Are we gonna use it?

Comment (aeioulisa) [reply dev-support/docker-images/jupyter/Dockerfile]: According to the description of [Jupyterlab](https://github.com/jupyterlab/jupyterlab-git),  it needs NodeJS to build the extension package.

**Review by pingsutw — APPROVED** [review]
LGTM

**Pull request closed without merging**